src/metpy/io/_tools.py

```diff
@@ -22,12 +22,13 @@ def open_as_needed(filename, mode='rb'):
     if hasattr(filename, 'read'):
         return filename
 
+    kwargs = {'errors': 'surrogateescape'} if mode != 'rb' else {}
```

Comment:
Since this is only used in the `else` clause with `open`, can you move the definition of `kwargs` there?

Reply:
done and done!